packages/salesforce-adapter/test/filters/add_missing_ids.test.ts

```diff
@@ -194,7 +195,25 @@ describe('Internal IDs filter', () => {
       expect(elements[2].annotations?.[INTERNAL_ID_ANNOTATION]).toBeUndefined()
       expect(elements[3].annotations?.[INTERNAL_ID_ANNOTATION]).toBeUndefined()
     })
-    it('should not run any query when feature is disabled', async () => {
+  })
+  describe('when feature is throwing an error', () => {
+    const mockListMetadataObjects: jest.Mock = jest.fn()
+    SalesforceClient.prototype.listMetadataObjects = mockListMetadataObjects
+    it('should return a warning', async () => {
+      const { connection } = mockClient()
+      connection.query.mockImplementation(() => { throw new Error() })
+      const res = await filter.onFetch(elements) as FilterResult
+      expect(res).toBeDefined()
+      expect(res.errors).toBeDefined()
+      const err = res.errors ?? []
+      expect(err[0].message).toEqual('unexpected error when adding missing ids')
```

Comment:
I think we can switch these 2 tests for 2 more concise tests.
Also we should verify that the error severity is warning. I think this is a 'property' that we would like the future developer to keep in line in the next PR's, therefore we should enforce it via a test.
```suggestion
      const err = res.errors ?? []
      expect(res.errors).toHaveLength(1)
      expect(err[0]).toEqual({
         severity: SaltoError.Warning,
         message: 'unexpected error when adding missing ids',
      })
```